cpp/include/cudf/hashing.hpp

```diff
@@ -38,6 +43,7 @@ namespace cudf {
 std::unique_ptr<column> hash(
   table_view const& input,
   hash_id hash_function                     = hash_id::HASH_MURMUR3,
+  retain_nulls policy                       = retain_nulls::HASH_NULLS,
```

Comment:
I'm wondering if this API really needs this option. Options in APIs inherently complicate a user's comprehension and the implementation and therefore should only be added when strictly necessary. In this case, if `hash` always returns a non-nullable result (i.e., it hashes null values), then the `RETAIN_NULLS` policy can be accomplished trivially by the user by just calling `bitmask_and` on the input table and updating the null mask on the result.

Reply:
Furthermore, adding this as an option requires that all code paths honor this option (i.e., the Murmur codepath would need to be updated too.

Reply:
I'll factor it out, makes some sense that the user modifies the resulting table. There's a hash value generated for the row either way, so modifying the column afterwards should be fine.

The most significant case is if we were to add functionality for an 'or' case -- where a null in any column of the row would create a null result. Skipping those rows could speed up the hash significantly, but that's not in the current scope of what we're trying to do.